File: core/entity/stat.go
package entity

import "fmt"

type Byte uint64

// @TODO: Move this to somewhere in the presentation layer
func (t Byte) String() string {
	if t < 1024 {
		return fmt.Sprintf("%dB", t)
	}

	count := 0
	floatBytes := float64(t)

	for floatBytes > 1024 {
		floatBytes = floatBytes / 1024
		count++
	}

	return fmt.Sprintf("%.2f %ciB", floatBytes, "KMGTPE"[count-1])
}

type Stat struct {
	ContainerID   ContainerID `json:"containerId"`
	CPUPercent    float64     `json:"cpuPercent"`
	MemoryPercent float64     `json:"memoryPercent"`
	MemoryUsage   Byte        `json:"memoryUsage"`
	MemoryTotal   Byte        `json:"memoryTotal"`
	NetworkInput  Byte        `json:"networkInput"`
	NetworkOutput Byte        `json:"networkOutput"`
	BlockRead     Byte        `json:"blockRead"`
	BlockWrite    Byte        `json:"blockWrite"`
}

File: web/core/lib/entity/Stat.ts
import { Byte } from './Byte';

export class Stat {
  constructor(obj: any) {
    this.cpuPercent = obj.cpuPercent;
    this.memoryPercent = obj.memoryPercent;
    this.memoryUsage = new Byte(obj.memoryUsage);
    this.memoryTotal = new Byte(obj.memoryTotal);
    this.networkInput = new Byte(obj.networkInput);
    this.networkOutput = new Byte(obj.networkOutput);
    this.blockRead = new Byte(obj.blockRead);
    this.blockWrite = new Byte(obj.blockWrite);
  }

  cpuPercent: number

  memoryPercent: number

  memoryUsage: Byte

  memoryTotal: Byte

  networkInput: Byte

  networkOutput: Byte

  blockRead: Byte

  blockWrite: Byte
}
